pkg/compact/compact.go

```diff
@@ -1036,10 +1062,14 @@ func (cg *Group) compact(ctx context.Context, dir string, planner Planner, comp
 		}
 		toCompactDirs = append(toCompactDirs, bdir)
 	}
+	blockSpan.Finish()
```

Comment:
Is it correct? The blockSpan is per block, right? Then you need to finish it after each block is processed. So it should be inside the for loop

Reply:
Sorry my bad. I've removed the block span now.